Improve Linux Install 2: Reorganize install/linux.md (#9082)
flutter/website: src/get-started/install/linux.md

|                                          | [Run flutter doctor](#run-flutter-doctor)                          |
| 3. [Update your path](#update-your-path) | [Temporarily update your path](#temporarily-update-your-path)      |
|                                          | [Permanently update your path](#permanently-update-your-path)      |
|                                          | [Update the path directly](#update-the-path-directly)              |
| 4. [Android setup](#android-setup)       | [Install Android Studio](#install-android-studio)                  |
|                                          | [Set up your Android device](#setup-android-device)                |
|                                          | [Set up the Android emulator](#setup-android-emulator)             |
|                                          | [Agree to Android licenses](#agree-android-licenses)               |
|                                          | [Install Google Chrome](#install-google-chrome)                    |
| 5. [Linux setup](#linux-setup)           | [Additional Linux requirements](#additional-linux-requirements)    |

> |                                          | [Understanding Linux Commands](#understanding-linux-commands)      |

Reviewer: **issue (blocking):** This section is generated automatically with the on-page Table of Contents. This should be removed.

  - `which`
  - `xz-utils`

>   - `zip`

Reviewer: **question:** Did you want to add links and install instructions as you did in [9081](https://github.com/flutter/website/pull/9081)?

> - **Shared libraries**: To utilize the Flutter `test` command, your environment needs the library `libGLU.so.1`. This is typically provided by mesa packages, such as `libglu1-mesa` on Ubuntu/Debian and `mesa-libGLU` on Fedora.
> 

Reviewer: **issue:** Use [semantic breaks](https://sembr.org/).

```suggestion
- **Shared libraries**: To utilize the Flutter `test` command, 
  your environment needs the library `libGLU.so.1`. 
  The mesa packages, like `libglu1-mesa` on Ubuntu/Debian and `mesa-libGLU` on Fedora,
  provide this library.
```
